src/rust_api.rs

```diff
@@ -0,0 +1,1561 @@
+    }
+
+    pub fn mov_left_shift(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::LShift)
+    }
+
+    pub fn mov_right_shift(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::RShift)
+    }
+
+    pub fn mov_negate(&mut self, arch: Arch) -> Move {
+        self.mov_internal(Source::Imm, arch, OpBits::Negate)
+    }
+
+    pub fn mov_mod(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::Mod)
+    }
+
+    pub fn mov_bit_xor(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::BitXor)
+    }
+
+    pub fn mov(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::NoOp)
+    }
+
+    pub fn mov_signed_right_shift(&mut self, source: Source, arch: Arch) -> Move {
+        self.mov_internal(source, arch, OpBits::SignRShift)
+    }
+
+    #[inline]
+    fn mov_internal(&mut self, source: Source, arch_bits: Arch, op_bits: OpBits) -> Move {
+        Move {
+            bpf_code: self,
+            src_bit: source,
+            op_bits: op_bits,
+            arch_bits: arch_bits,
+            insn: Insn {
+                opc: 0x00,
+                dst: 0x00,
+                src: 0x00,
+                off: 0x00_00,
+                imm: 0x00_00_00_00
+            }
+        }
+    }
+
+    pub fn swap_bytes(&mut self, endian: Endian) -> SwapBytes {
+        SwapBytes {
+            bpf_code: self,
+            endian: endian,
+            insn: Insn {
+                opc: 0x00,
+                dst: 0x00,
+                src: 0x00,
+                off: 0x00_00,
+                imm: 0x00_00_00_00
+            }
+        }
+    }
+
+    pub fn load(&mut self, mem_size: MemSize) -> Load {
+        self.load_internal(mem_size, Addressing::Undef, 0x00)
+    }
+
+    pub fn load_abs(&mut self, mem_size: MemSize) -> Load {
+        self.load_internal(mem_size, Addressing::Abs, 0x00)
+    }
+
+    pub fn load_ind(&mut self, mem_size: MemSize) -> Load {
+        self.load_internal(mem_size, Addressing::Ind, 0x00)
+    }
+
+    pub fn load_x(&mut self, mem_size: MemSize) -> Load {
+        self.load_internal(mem_size, Addressing::Undef, 0x61)
+    }
+
+    #[inline]
+    fn load_internal(&mut self, mem_size: MemSize, addressing: Addressing, source: u8) -> Load {
+        Load {
+            bpf_code: self,
+            addressing: addressing,
+            mem_size: mem_size,
+            source: source,
+            insn: Insn {
+                opc: 0x00,
+                dst: 0x00,
+                src: 0x00,
+                off: 0x00_00,
+                imm: 0x00_00_00_00
+            }
+        }
+    }
+
+    pub fn store(&mut self, mem_size: MemSize) -> Store {
+        self.store_internal(mem_size, 0x00)
+    }
+
+    pub fn store_x(&mut self, mem_size: MemSize) -> Store {
+        self.store_internal(mem_size, 0x61)
```

Comment:
Nope, I didn't have hallucinations yesterday 😄. Here, I didn't know what constant I should use. `BPF_MEM | BPF_STX` works.